dotnet/src/SemanticKernel.Abstractions/SkillDefinition/FunctionsView.cs

```diff
@@ -94,22 +77,120 @@ public bool IsSemantic(string skillName, string functionName)
     /// <exception cref="AmbiguousMatchException"></exception>
     public bool IsNative(string skillName, string functionName)
     {
-        var sf = this.SemanticFunctions.ContainsKey(skillName)
-                 && this.SemanticFunctions[skillName]
-                     .Any(x => string.Equals(x.Name, functionName, StringComparison.OrdinalIgnoreCase));
+        return this.IsFunctionCheck(skillName, functionName, nativeFunctionCheck: true);
+    }
+
+#pragma warning disable CA1859 // Use concrete types when possible for improved performance
+    private IReadOnlyDictionary<string, IReadOnlyCollection<FunctionView>> GetFunctions(bool nativeFunctionCheck)
+#pragma warning restore CA1859 // Use concrete types when possible for improved performance
+    {
+        ConcurrentDictionary<string, List<FunctionView>> targetFunctionDictionary = (nativeFunctionCheck)
+            ? this._nativeFunctions
+            : this._semanticFunctions;
+
+        return new ReadOnlyFunctionsView(targetFunctionDictionary);
+    }
+
+    private class ReadOnlyFunctionsView : IReadOnlyDictionary<string, IReadOnlyCollection<FunctionView>>
```

Comment:
Classes `ReadOnlyFunctionsView` and `ReadOnlyFunctionsViewEnumerator` are nested, but they don't use any members from parent class. Also, they are located between `FunctionsView` class methods, which makes them harder to read. Can we move these classes to separate files and make them `internal`?

Reply:
They are not used outside of the FunctionsView class, might keep it inside a region and move the private property before them.